api/src/organizational-entities/domain/usecases/update-organizations-in-batch.usecase.js

```diff
@@ -0,0 +1,84 @@
+import { createReadStream } from 'node:fs';
+
+import bluebird from 'bluebird';
+
+import { DomainTransaction } from '../../../shared/domain/DomainTransaction.js';
+import { CsvColumn } from '../../../shared/infrastructure/serializers/csv/csv-column.js';
+import { CsvParser } from '../../../shared/infrastructure/serializers/csv/csv-parser.js';
+import { getDataBuffer } from '../../../shared/infrastructure/utils/buffer.js';
+import { OrganizationBatchUpdateDTO } from '../dtos/OrganizationBatchUpdateDTO.js';
+
+const CSV_HEADER = {
+  columns: [
+    new CsvColumn({
+      isRequired: true,
+      name: 'Organization ID',
+      property: 'id',
+    }),
+    new CsvColumn({
+      name: 'Organization Name',
+      property: 'name',
+    }),
+    new CsvColumn({
+      name: 'Organization External ID',
+      property: 'externalId',
+    }),
+    new CsvColumn({
+      name: 'Organization Parent ID',
+      property: 'parentOrganizationId',
+    }),
+    new CsvColumn({
+      name: 'Organization Identity Provider Code',
+      property: 'identityProviderForCampaigns',
+    }),
+    new CsvColumn({
+      name: 'Organization Documentation URL',
+      property: 'documentationUrl',
+    }),
+    new CsvColumn({
+      name: 'Organization Province Code',
+      property: 'provinceCode',
+    }),
+    new CsvColumn({
+      name: 'DPO Last Name',
+      property: 'dataProtectionOfficerLastName',
+    }),
+    new CsvColumn({
+      name: 'DPO First Name',
+      property: 'dataProtectionOfficerFirstName',
+    }),
+    new CsvColumn({
+      name: 'DPO E-mail',
+      property: 'dataProtectionOfficerEmail',
+    }),
+  ],
+};
+
+/**
+ * @typedef {function} updateOrganizationsInBatch
+ * @param {Object} params
+ * @param {string} params.filePath
+ * @param {OrganizationForAdminRepository} params.organizationForAdminRepository
+ * @return {Promise<void>}
+ */
+export const updateOrganizationsInBatch = async function ({ filePath, organizationForAdminRepository }) {
+  const organizationBatchUpdateDtos = await _getCsvData(filePath);
+
+  if (organizationBatchUpdateDtos.length === 0) return;
+
+  await DomainTransaction.execute(async (domainTransaction) => {
+    await bluebird.mapSeries(organizationBatchUpdateDtos, async (organizationBatchUpdateDto) => {
+      const organization = await organizationForAdminRepository.get(organizationBatchUpdateDto.id, domainTransaction);
+      organization.updateFromOrganizationBatchUpdateDto(organizationBatchUpdateDto);
+      await organizationForAdminRepository.update(organization, domainTransaction);
+    });
```

Comment:
Pourquoi utiliser une exécution en série des promesses ici ? Chaque update du mass update est indépendant des autres n'est-ce pas ?

Reply:
À la réflexion je pense que j'ai trouvé la réponse : c'est pour ne pas imposer une trop grosse charge à l'API si il y a plusieurs milliers d'orga à modifier en masse, un Promise.all ralentirait beaucoup le système.

Reply:
C'est vrai que je devrais plutôt le faire en parallèle du faite que l'on utilise une transaction pour la liste. Je vais faire le changement.

Reply:
> À la réflexion je pense que j'ai trouvé la réponse : c'est pour ne pas imposer une trop grosse charge à l'API si il y a plusieurs milliers d'orga à modifier en masse, un Promise.all ralentirait beaucoup le système.

Normalement on n'aura pas beaucoup d'organisations dans les listes. On verra si un _Promise.all_ pose problème.

Reply:
On peut aussi mettre une boucle `for...in` qui permet d'exécuter des promesses en série sans utiliser la bibliothèque *bluebird* qui est en fin de vie.